qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -109,6 +109,9 @@ def _opt_control(property_set):
             Optimize1qGates(), CommutativeCancellation(),
             OptimizeSwapBeforeMeasure(), RemoveDiagonalGatesBeforeMeasure()]
 
+    if not coupling_map:
+        _opt.pop(4)  # if no coupling map, don't call the CXDirection pass
```

Comment:
This is very much dependant on how the the array `_opt` is constructed. I think the best is not to added instead of removing it.

Reply:
I agree.

Reply:
So something like 
```
    _opt =  [RemoveResetInZeroState(),
            Collect2qBlocks(), ConsolidateBlocks(),
            Unroller(basis_gates), Optimize1qGates(), CommutativeCancellation(),
            OptimizeSwapBeforeMeasure(), RemoveDiagonalGatesBeforeMeasure()]

    if coupling_map:
        _opt.append(CXDirection(coupling_map))  # unroll unitaries and match coupling
```
would be enough, or is it necessary to preserve the order of the passes ? (I'm not really sure about that)

Reply:
Yes that would work. Since the whole blocks runs iteratively until convergence, I don't think the order matters much.

Reply:
ok, its corrected then

Reply:
Thanks it looks good to me now.